internal/core/items/text.rs

```diff
@@ -970,6 +967,10 @@ impl TextInput {
         window_adapter: &Rc<dyn WindowAdapter>,
         self_rc: &ItemRc,
     ) {
+        self.preedit_text.set(Default::default());
+        self.preedit_selection_start.set(0);
+        self.preedit_selection_end.set(0);
```

Comment:
This might or might not help, but it carries the risk of the input method being out of sync with the state that we think it is in. When the text of the input field changes, we should inform the input method about what the new text is. Qt has APIs for that, winit unfortunately doesn't.